File: spec/python/test_bcd_user_type_le.py
import unittest

from bcd_user_type_le import BcdUserTypeLe

class TestBcdUserTypeLe(unittest.TestCase):
    def test_bcd_user_type_le(self):
        r = BcdUserTypeLe.from_file("src/bcd_user_type_le.bin")

        self.assertEqual(r.ltr.as_int, 12345678)
        self.assertEqual(r.ltr.as_str, "12345678")
        self.assertEqual(r.rtl.as_int, 87654321)
        self.assertEqual(r.rtl.as_str, "87654321")
        self.assertEqual(r.leading_zero_ltr.as_int, 123456)
        self.assertEqual(r.leading_zero_ltr.as_str, "00123456")

File: spec/java/src/io/kaitai/struct/spec/TestBcdUserTypeLe.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.BcdUserTypeLe;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestBcdUserTypeLe extends CommonSpec {
    @Test
    public void testBcdUserTypeLe() throws Exception {
        BcdUserTypeLe r = BcdUserTypeLe.fromFile(SRC_DIR + "bcd_user_type_le.bin");

        assertEquals(r.ltr().asInt().intValue(), 12345678);
        assertEquals(r.ltr().asStr(), "12345678");
        assertEquals(r.rtl().asInt().intValue(), 87654321);
        assertEquals(r.rtl().asStr(), "87654321");
        assertEquals(r.leadingZeroLtr().asInt().intValue(), 123456);
        assertEquals(r.leadingZeroLtr().asStr(), "00123456");
    }
}
